Feature: Add "toggle coordinates" button in Puzzles

melissa-z opened the pull request:
Fixes #1454
## Proposed Changes

Hi, first time contributing. Love OGS! 

  - Add a "toggle coordinates" button in Puzzles
  - Add `title` attribute to buttons
  - Minor changes to clean up button styling 
     - Remove border outline and add margin instead (feels more modern and the spacing improves accessibility)
     - Fix active state border color

![Screen Recording 2021-04-18 at 2 19 52 PM](https://user-images.githubusercontent.com/34780520/115161273-a7074a00-a051-11eb-849e-c7fdf4b0d903.gif)

<img width="366" alt="Screen Shot 2021-04-18 at 2 20 43 PM" src="https://user-images.githubusercontent.com/34780520/115161266-98b92e00-a051-11eb-8234-571a5b52a885.png">

benjaminpjones commented:
Awesome! One question: did you consider using the existing coordinates icon for the button?  To me it feels like the full "Toggle Coordinates" takes up a lot of space compared to the other buttons.

![image](https://user-images.githubusercontent.com/25233703/115130217-b471f500-9fbb-11eb-8ccb-7ea88f4b7a3c.png)

anoek commented:
https://github.com/online-go/online-go.com/blob/devel/src/views/Game/Game.tsx#L3143 shows the usage, you should just be able to drop in `<i className="ogs-coordinates" />` for that icon I think

melissa-z commented:
That's a great point, will do!

melissa-z commented:
I switched out "Toggle Coordinates" for the icon, and also added some `title` attributes to the icon-only buttons so that a text description will appear on hover. Any thoughts?

benjaminpjones commented:
Looks nice! Thanks!  And yeah I think the tooltips are helpful.

benjaminpjones commented on src/views/Puzzle/Puzzle.tsx:
```diff
@@ -779,6 +779,7 @@ export class Puzzle extends React.Component<PuzzleProperties, PuzzleState> {
             <div className="btn-container">
                 <div className="btn-group">
                     <button type="button"
+                        title="Flip Diagonal"
```
What I meant in the other comment:

```title={_("Flip diagonal")}```

Alternatively, if you want to add some context:

```title={pgettext("Puzzle options", "Flip Diagonal")```

The translators will be able to see this in their dashboard and add a translation later.

melissa-z replied:
Fixed!

benjaminpjones replied:
Cool, thanks!

anoek commented:
Thanks for the improvements!
